modules/mono/editor/GodotTools/GodotTools/Ides/MessagingServer.cs

```diff
@@ -385,9 +385,12 @@ private static async Task<Response> HandleCodeCompletionRequest(CodeCompletionRe
                 // However, it doesn't fix resource loading if the rest of the path is also case insensitive.
                 string scriptFileLocalized = FsPathUtils.LocalizePathWithCaseChecked(request.ScriptFile);
 
+                // The node API can only be called from the main thread.
+                await Godot.Engine.GetMainLoop().ToSignal(Godot.Engine.GetMainLoop(), "process_frame");
```

Comment:
A bit hacky but we could wrap this line in an if so it only awaits when autocompleting NodePaths.

Also nit-pick, we should probably avoid calling `GetMainLoop` twice. And the signal name can use the exposed `StringName`.

```suggestion
                if (request.Kind == CodeCompletionRequest.CompletionKind.NodePaths)
                {
                    var mainLoop = Godot.Engine.GetMainLoop();
                    await mainLoop.ToSignal(mainLoop, Godot.SceneTree.SignalName.ProcessFrame);
                }
```

By the way, since the signal is in `SceneTree` and not `MainLoop`, this would block forever if the main loop is not a `SceneTree`. Although I don't think the main loop would ever change in the Godot editor.

Reply:
We don't want the method to block waiting for the completion. That's probably why it was using `Task.Run` before. So, I just decided to always await the signal, even if not needed.

Reply:
Oh yeah, then maybe we could `await Task.CompletedTask` in the else block.